app/views/posts/index.html.erb

```diff
@@ -3,9 +3,21 @@
     <h1 class="text-4xl font-bold">Posts</h1>
     <%= link_to 'New post', new_post_path, class: 'block btn-to-submit' %>
   </div>
-  <div id="posts" class="min-w-full">
-    <% @posts.each do |post| %>
-      <%= render 'preview', post: post %>
+  <%= turbo_frame_tag "#{@offset}offset", target: "#{@offset += 1}offset" do %>
+    <div id="posts" class="min-w-full">
+      <% @posts.each do |post| %>
+        <%= render 'preview', post: post %>
+      <% end %>
+    </div>
+    <% if @has_more_posts %>
+      <div class="flex items-center min-w-full">
```

Comment:
This provides "infinite scrolling" but the continuity will be messed up if some new posts appear.

Reply:
There is also the chance of having all those 16 posts invisible to the user so they get nothing after clicking "Show more."